src/pevm.rs

```diff
@@ -231,99 +231,136 @@ impl Pevm {
+                                    max_fee +=
+                                        U256::from(tx.get_total_blob_gas()) * U256::from(blob_fee);
+                                }
+                                if balance < max_fee {
+                                    return Err(PevmError::ExecutionError(
+                                        "Transaction(LackOfFundForMaxFee)".to_string(),
+                                    ));
+                                }
+                                balance -= addition;
+                                // End of overflow TODO
+
+                                nonce += 1;
                             }
-                            balance -= addition;
-                            // End of overflow TODO
+                            // TODO: Better error handling
+                            _ => unreachable!(),
+                        }
 
-                            nonce += 1;
+                        // SAFETY: The multi-version data structure should not leak an index over block size.
+                        let tx_result =
+                            unsafe { fully_evaluated_results.get_unchecked_mut(*tx_idx) };
+                        let account = tx_result.state.entry(address).or_default();
+                        // TODO: Deduplicate this logic with [PevmTxExecutionResult::from_revm]
+                        if spec_id.is_enabled_in(SPURIOUS_DRAGON)
+                            && code_hash.is_none()
+                            && nonce == 0
+                            && balance == U256::ZERO
+                        {
+                            *account = None;
+                        } else if let Some(account) = account {
+                            // Explicit write: only overwrite the account info in case there are storage changes
+                            // TODO: Can code be changed mid-block?
+                            account.balance = balance;
+                            account.nonce = nonce;
+                        } else {
+                            // Implicit write: e.g. gas payments to the beneficiary account,
+                            // which doesn't have explicit writes in [tx_result.state]
+                            *account = Some(EvmAccount {
+                                balance,
+                                nonce,
+                                code_hash,
+                                code: code.clone(),
+                                storage: AHashMap::default(),
+                            });
                         }
-                        // TODO: Better error handling
-                        _ => unreachable!(),
                     }
+                }
+                MemoryLocation::CodeHash(_) => unreachable!(),
+                MemoryLocation::Storage(address, index) => {
+                    let mut storage_value = U256::ZERO;
+                    if let Ok(value) = storage.storage(&address, &index) {
```

Comment:
We can use the same micro-optimization like in basic here: do not read from storage if the first write is an absolute value.